rt/transports/http/src/main/java/org/apache/cxf/transport/http/HTTPConduit.java

```diff
@@ -1600,7 +1605,7 @@ protected int doProcessResponseCode() throws IOException {
             }
             if (exchange != null) {
                 exchange.put(Message.RESPONSE_CODE, rc);
-                if (rc == 404 || rc == 503 || rc == 429) {
+                if ((rc == 404 && MessageUtils.getContextualBoolean(outMessage, SERVICE_NOT_AVAILABLE_ON_HTTP_404, true)) || rc == 503 || rc == 429) {
```

Comment:
Thanks @kwin , I am wondering if we could use list of status codes as contextual property instead (`404`, `503`, `429` being the default)?

Reply:
Makes sense, I am gonna change the PR

Reply:
Done in https://github.com/apache/cxf/pull/964/commits/191bea69cff1ff282d6565d0ba1e4b49daa6d95b